src/Generics/SOP/BasicFunctors.hs

```diff
@@ -61,6 +75,54 @@ instance Traversable (K a) where
   traverse _ (K x) = pure (K x)
 #endif
 
+#ifdef LIFTED_CLASSES
+instance Eq2 K where
+    liftEq2 eq _ (K x) (K y) = eq x y
+instance Ord2 K where
+    liftCompare2 comp _ (K x) (K y) = comp x y
+instance Read2 K where
+    liftReadsPrec2 rp _ _ _ = readsData $
+         readsUnaryWith rp "K" K
+instance Show2 K where
+    liftShowsPrec2 sp _ _ _ d (K x) = showsUnaryWith sp "K" d x
+
+instance (Eq a) => Eq1 (K a) where
+    liftEq = liftEq2 (==)
+instance (Ord a) => Ord1 (K a) where
+    liftCompare = liftCompare2 compare
+instance (Read a) => Read1 (K a) where
+    liftReadsPrec = liftReadsPrec2 readsPrec readList
+instance (Show a) => Show1 (K a) where
+    liftShowsPrec = liftShowsPrec2 showsPrec showList
+#else
+instance (Eq a) => Eq1 (K a) where
+    eq1 (K x) (K y) = x == y
+instance (Ord a) => Ord1 (K a) where
+    compare1 (K x) (K y) = compare x y
+instance (Read a) => Read1 (K a) where
+    readsPrec1 = readsData $ readsUnary "K" K
+instance (Show a) => Show1 (K a) where
+    showsPrec1 d (K x) = showsUnary "K" d x
+#endif
+
+-- This have to be implemented manually
```

Comment:
I don't remember anymore, have to recheck, but I guess because it mono-kindize `K`.